<?xml version='1.0'?>
<ns2:test-suite xmlns:ns2='urn:model.allure.qatools.yandex.ru' start='1735878985674' stop='1735878988969'>
    <name>API_Delete Books</name>
    <title>API_Delete Books</title>
    <test-cases>
        <test-case start='1735878986018' status='passed' stop='1735878986229'>
            <name>Valid Case: Update a book with valid parameters</name>
            <title>Valid Case: Update a book with valid parameters</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
        </test-case>
        <test-case start='1735878986253' status='failed' stop='1735878986934'>
            <name>Invalid Case: Delete a book with an invalid ID format</name>
            <title>Invalid Case: Delete a book with an invalid ID format</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>expected '' to equal 'Invalid Input Parameters.'</message>
                <stack-trace>AssertionError: expected '' to equal 'Invalid Input Parameters.'
                    at Context.eval (webpack://itqa/./cypress/e2e/api/DeleteBook.cy.js:58:37)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1735878987021' status='failed' stop='1735878987690'>
            <name>Authorization Case: Delete a book without authorization headers</name>
            <title>Authorization Case: Delete a book without authorization headers</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>expected '' to equal 'You are not authorized to delete the book.'</message>
                <stack-trace>AssertionError: expected '' to equal 'You are not authorized to delete the book.'
                    at Context.eval (webpack://itqa/./cypress/e2e/api/DeleteBook.cy.js:70:37)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1735878987741' status='broken' stop='1735878988215'>
            <name>Forbidden Case: Delete a book with insufficient permissions</name>
            <title>Forbidden Case: Delete a book with insufficient permissions</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>userAuthHeader is not defined</message>
                <stack-trace>ReferenceError: userAuthHeader is not defined
                    at Context.eval (webpack://itqa/./cypress/e2e/api/DeleteBook.cy.js:78:21)</stack-trace>
            </failure>
        </test-case>
        <test-case start='1735878988279' status='failed' stop='1735878988937'>
            <name>Invalid Case: Delete a book with a non-existent ID</name>
            <title>Invalid Case: Delete a book with a non-existent ID</title>
            <labels/>
            <parameters/>
            <steps/>
            <attachments/>
            <failure>
                <message>expected 403 to equal 404</message>
                <stack-trace>AssertionError: expected 403 to equal 404
                    at Context.eval (webpack://itqa/./cypress/e2e/api/DeleteBook.cy.js:95:39)</stack-trace>
            </failure>
        </test-case>
    </test-cases>
</ns2:test-suite>